### ISSUE TITLE
about the text feature

### ISSUE THREAD

**Issue by soohyunpyo** [zhengli97/PromptKD#33]
hello im very interested in your paper and want to reproduce and want to apply it to my method.
And i have a question for your code. in your paper and Pseudocode , you wrote that you use precomputed text feature but in your code, the VLPromptLearner's forward function is executed in CustomCLIP_teacher's forward function at every step. I think this part significantly prolongs the training time. I modified it like Pseudocode, but its performance(75.2,70.8) is so lower than original implementation' performance(79.1,73.6) with big margin. I performed it for imagenet. I would be so appreciated if you give me some opinion for it.
Thank you for your good paper and implementation code.

**Comment by zhengli97** [comment]
If you don't modify the code, can you achieve the accuracy reported in the paper?

**Comment by soohyunpyo** [comment]
Yes, I have achieved the performance described in your paper using your code, and I have resolved the issue. After modifying it like pseudocode, I realized that the text features generated during the initialization phase were meaningless before the weights were loaded. By updating the text features after loading the weights during initialization, I was able to achieve the same results with approximately 5 times faster training time. Thank you for your response and good paper. And I hope this can also be helpful for your code.

**Comment by zhengli97** [comment]
Here is some information for researchers who want to implement this method: The way @soohyunpyo does this provides a new idea that everyone can consider implementing. In my own implementation, I achieve this operation as follows: after the teacher pre-training is completed, save the well-trained text features in .npy format, and then directly load this file at the beginning of the student training, convert it into a tensor and use it directly. If researchers want to achieve this kind of acceleration operation, both of the above methods can be considered.
